### PR TITLE
apmbench: Replace -warmup-events for -warmup-time

### PR DESCRIPTION
**Pull request by marclop**
## Motivation/summary

Removes the `-warmup-events` flag and the complicated logic required to
handle timeouts it in favor of a new flag `-warmup-time` which will send
events for the specified duration. This strategy falls more in line with
the intended usage for a warmup phase, where a user wants to warm up the
target server for an amount of time using the specified agents and does
not care much about how many events are sent for that warmup phase.

Also re-formats some unformatted tf files, and adds a new make benchmark
target which auto-tunes the number of agents from the APM Server size.

Last, it changes apmbench to not return an error when the warm-up phase
returns an error. Warm-ups are best-effort and shouldn't impede the
benchmarks from running.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
- [x] Documentation has been updated

## How to test these changes

Run `apmbench` with `-warmup-time` and see if it warms up the APM Server for the set time.

## Related issues

Part of #8775

### CODE REVIEW

File: systemtest/benchtest/main_test.go [hunk 1]
```diff
@@ -37,166 +37,56 @@ import (
...
-					}
-
-					if !strings.HasPrefix(r.URL.Path, "/intake") {
+		t.Run(fmt.Sprintf("%d_agent_%s_events", c.agents, c.duration.String()), func(t *testing.T) {
+			var received uint64
+			srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if r.URL.Path == "/debug/vars" {
+					// Report idle APM Server.
+					w.Write([]byte(`{"libbeat.output.events.active":0}`))
+				}
```
Comment (kruskall) [inline systemtest/benchtest/main_test.go:54]: **issue**: we should return early here. if `w.Write` is called the server has already sent a response to the client

File: systemtest/benchtest/main.go [hunk 2]
```diff
@@ -265,6 +256,9 @@ func warmup(agents int, events uint, url, token string) error {
 	var merr multierror.Errors
 	for err := range errs {
 		if err != nil && !errors.Is(err, context.DeadlineExceeded) {
+			if e, ok := err.(timeoutError); ok && e.Timeout() {
```
Comment (kruskall) [inline systemtest/benchtest/main.go:259]: **suggestion:** I think we can use `errors.As` and `net.Error` to check if `err` is a network error and then call `Timeout()` on it.
We should also add a TODO to remove this once we update to Go 1.19 as the workaround is no longer necessary: https://go.dev/doc/go1.19#net

File: systemtest/benchtest/main_test.go [hunk 3]
```diff
@@ -37,166 +37,56 @@ import (
...
-		t.Run(tt.name, func(t *testing.T) {
-			got := warmupTimeout(
-				tt.args.ingestRate,
-				tt.args.events,
-				tt.args.epm,
-				tt.args.agents,
-				tt.args.cpus,
-			)
-			assert.Equal(t, tt.expected, got)
+				}
```
Comment (kruskall) [inline systemtest/benchtest/main_test.go:82]: **issue:** `scanner.Err()` should be checked before returning status accepted

**Review by kruskall — CHANGES_REQUESTED** [review]

**Comment by marclop** [comment]
@kruskall thanks for the suggestions. Addressed them in the last commit.

**Review by kruskall — APPROVED** [review]
LGTM!

**Pull request merged**